packages/strapi-design-system/src/DismissableLayer/DimissableLayer.js

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+export const DismissableLayer = ({ children, className, onEscapeKeyDown, onPointerDownOutside }) => {
+  const layerRef = React.useRef(null);
+  const onEscapeKeyDownHandler = useCallbackRef(onEscapeKeyDown);
+  const onPointerDownOutsideHandler = useCallbackRef(onPointerDownOutside);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').KeyboardEventHandler<Document>}
+     */
+    const handleKeyDown = (event) => {
+      if (event.key === 'Escape') {
+        onEscapeKeyDownHandler(event);
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, [onEscapeKeyDownHandler]);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').PointerEventHandler<Document>}
+     */
+    const handlePointerDownOutside = (event) => {
+      if (layerRef.current && !layerRef.current.contains(event.target)) {
+        onPointerDownOutsideHandler();
+      }
+    };
+
+    document.addEventListener('pointerdown', handlePointerDownOutside);
+
+    return () => document.removeEventListener('pointerdown', handlePointerDownOutside);
+  }, [onPointerDownOutsideHandler]);
+
+  return (
+    <div ref={layerRef} className={className}>
+      {children}
+    </div>
+  );
+};
+
+DismissableLayer.propTypes = {
+  children: PropTypes.node,
```

Comment:
It could be, my view on required is:
* Does it need a default?
* Will the component break without it
So i'd say it doesn't have to be, but if you think that's more inline with other components then it can be?

Reply:
Yeah I think it would be beneficial to see from the prop-types that the component should have children and I think that is the way we handle it elsewhere in the DS.